core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -45,31 +50,67 @@ protected FileScanTask self() {
 
   @Override
   protected FileScanTask newSplitTask(FileScanTask parentTask, long offset, long length) {
-    return new SplitScanTask(offset, length, parentTask);
+    return new SplitScanTask(offset, length, deletesSizeBytes(), parentTask);
```

Comment:
This ensures the size of deletes is only computed once for all split tasks generated from the same file task.